### PR TITLE
Feature/#1versionconflicts

### PR DESCRIPTION
**Pull request by yasserf**
Not finished, raising for Tests

### CODE REVIEW

File: src/constants/merge-strategies.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+function merge( oldData, newData, callback ) {
+	var prop;
+	for( prop in newData ) {
+		if( typeof( newData[ prop ] ) === "object" ) {
+			if( !prop in oldData || typeof( oldData[ prop ] ) !== "object" ) {
+				// Safe override or addition
+				oldData[ prop ]= newData[ prop ]; 
+			} else {
+				// Recursive
+				oldData[ prop ] = this.merge( oldData[ prop ], newData[ prop ] );
```
Comment (WolframHempel) [inline src/constants/merge-strategies.js]: do we need the `this`?

File: src/constants/merge-strategies.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+function merge( oldData, newData, callback ) {
```
Comment (WolframHempel) [inline src/constants/merge-strategies.js]: guess I'm missing something, but where's the no conflict bit?

File: src/record/record.js [hunk 3]
```diff
@@ -264,7 +286,7 @@ Record.prototype._$onMessage = function( message ) {
 		this._applyUpdate( message, this._client );
 	}
 	else if( message.data[ 0 ] === C.EVENT.VERSION_EXISTS ) {
-		this._recoverRecord( message );
+		this._recoverRecord( message.data[ 2 ], JSON.parse( message.data[ 3 ] ), message );
```
Comment (WolframHempel) [inline src/record/record.js]: Could we used the `typed` mechanism for that

Comment (yasserf) [inline src/record/record.js]: Object being sent here isn't using typed since it is always JSON, same as read and update

**Pull request closed without merging**